examples/gelu/src/lib.rs

```diff
@@ -1,20 +1,15 @@
 use cubecl::prelude::*;
 
-#[derive(CubeLaunch)]
-struct Input<F: Float> {
-    inputs: Sequence<Array<F>>,
-}
-
 #[cube(launch_unchecked)]
-fn gelu_array<F: Float>(input: &Array<F>, output: &mut Array<F>) {
+fn gelu_array<F: Float>(input: &Array<Line<F>>, output: &mut Array<Line<F>>) {
     if ABSOLUTE_POS < input.len() {
         output[ABSOLUTE_POS] = gelu_scalar::<F>(input[ABSOLUTE_POS]);
     }
 }
 
 #[cube]
-fn gelu_scalar<F: Float>(x: F) -> F {
-    x * (F::erf(x / F::new(2.0f32.sqrt())) + F::new(1.0)) / F::new(2.0)
+fn gelu_scalar<F: Float>(x: Line<F>) -> Line<F> {
+    x * (Line::erf(x / Line::cast_from(2.0f32.sqrt())) + 1.0) / 2.0
```

Comment:
Here it's a bit weird no? I get it that the generic F of Line::<F>::erf or Line::<F>::cast_from can be inferred, but it feels weird, because you don't cast sqrt(2) to a line, you cast it to F.